llvm/utils/TableGen/DecoderEmitter.cpp

```diff
@@ -1950,21 +1959,13 @@ static void addOneOperandFields(const Record *EncodingDef, const BitsInit &Bits,
   }
 }
 
-static unsigned populateInstruction(
-    const CodeGenTarget &Target, const Record *EncodingDef,
-    const CodeGenInstruction *Inst, unsigned EncodingID,
-    std::map<unsigned, std::vector<OperandInfo>> &EncodingOperands,
-    bool IsVarLenInst) {
+unsigned InstructionEncoding::populateEncoding() {
+  bool IsVarLenInst = isa<DagInit>(EncodingDef->getValueInit("Inst"));
```

Comment:
There is a small difference here. Previously, the value of this flag was `Target.hasVariableLengthEncodings()`, now we determine it based on the type of `Inst` field -- if it is a DAG, then this must be a variable-length encoding.

(I'm not sure we need the global `hasVariableLengthEncodings` bit at all. We could freely mix variable and fixed-length encodings.)

Reply:
This seems reasonable. For M68k, I am assuming all insts have DAG for Inst, so they will all get this flag as true.